src/SwiftTaskProvider.ts

```diff
@@ -40,9 +40,10 @@ interface TaskConfig {
  * This excludes test targets.
  */
 function createBuildAllTask(): vscode.Task {
+    const additionalArgs = (process.platform !== 'darwin') ? ['--enable-test-discovery'] : [];
```

Comment:
What version of Swift do we want to support? I'm in favour of 5.5 and up which means test discovery is on by default

Reply:
There is an issue with the automatic test discovery where if you have an empty `LinuxMain.swift` it doesn't work. But you can't remove the `LinuxMain.swift` if your library supports earlier versions of swift.

Also I don't see a reason for not supporting earlier versions of swift.

Reply:
Personally to avoid complexity and warnings I'd be in favour of ignoring `LinuxMain.swift` was ever a thing but happy to defer

Reply:
@0xTim libraries that support older versions of swift (eg Soto) need so tests can run on those versions of swift

Reply:
Soto doesn't support less than 5.1 right? So test discovery will always be available to anyone running it. (This is all hypothetical at this point anyway as I'm happy to merge as the PR is)

Reply:
no it doesn't :-)